### ISSUE TITLE
`no-duplicates` false positive and autofix creates a syntax error

### ISSUE THREAD

**Issue by NickHeiner** [import-js/eslint-plugin-import#2114]
From:

```ts
import type { AccountQuery } from '@netflix-internal/dnajs';
import type dnajs from '@netflix-internal/dnajs';
```

To:

```ts
import dnajs, type { AccountQuery } from '@netflix-internal/dnajs';
```

The proper autocorrect is:

```ts
import type {default as dnajs, AccountQuery} from '@netflix-internal/dnajs';
```

On version `2.20.0` (**edit: and `2.23.4`**)

Another bug:

From: 
```ts
import type Registration from './Registration';
import type { RouteResolver } from './Registration';
```

To: 
```ts
import type Registration, { RouteResolver } from './Registration';
```

This gives TS error:

```
A type-only import can specify a default import or named bindings, but not both.ts(1363)
```

**Comment by ljharb** [comment]
@NickHeiner that's an old version, can you try on the latest?

**Comment by NickHeiner** [comment]
Ha! Blazing fast response. I literally was doing that. :smile:

**Comment by ljharb** [comment]
Thanks, that's definitely a bug.

Although I assume your OP example code is partially edited - can you provide the *exact* code that's failing, no "foo" or "bar"?

**Comment by NickHeiner** [comment]
🤦 lol, if I was going to edit the input, I should have edited the output as well.

I've updated the OP to be unedited. Sorry for the churn.

**Comment by ljharb** [comment]
Should the first one be `import type dnajs, { AccountQuery } from '@netflix-internal/dnajs';`? or can those not be combined with `import type` syntax?

**Comment by ljharb** [comment]
looks like in TypeScript, `import type` has to keep default and named separate. I'm not sure if Flow has the same limitation.

**Comment by NickHeiner** [comment]
Yes, I think an `import type` and `import` statement need to be separate.

**Comment by ljharb** [comment]
@NickHeiner yes but in this case it's two `import type` statements that can't be combined, despite the equivalent `import` statements being combineable.

**Comment by NickHeiner** [comment]
Right. Also, I found what the right edit should be, and updated the OP to reflect it.

**Comment by ljharb** [comment]
While `{ default as ` works, I do not think that's the proper autocorrect - i would never want to have `default as` in my codebase. I think those should just be left as two separate statements.

**Comment by NickHeiner** [comment]
Heh, fair. I'm happy to have `default as` rather than `eslint-disable import/no-duplicates` for now. :smile: So maybe this is something that should be behind an option.

**Comment by ljharb** [comment]
I think you can manually fix it to that - the lint rule wouldn't complain - so no option would be needed. eslint rules don't tend to have options that solely govern autofix behavior.

**Comment by GoodForOneFare** [comment]
If someone has time for review, #2149 should fix this. It's not exactly the solution proposed here, but is hopefully good enough 🙏
